### PR TITLE
Add inline spinner variant

### LINKED ISSUE #2533: Add inline spinner variant
The inline spinner variant referenced at https://www.lightningdesignsystem.com/components/spinners/#Inlined is currently not supported. This is necessary to properly implement the combobox menu loading state as mentioned at https://www.lightningdesignsystem.com/components/combobox/#Loading-more-options.

I'm planning to contribute this myself shortly. This relates to W-7747704.

### PR DESCRIPTION
**Pull request by jimmydief**
Fixes #2533.

### Additional description

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [ ] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [ ] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

**Comment by jimmydief** [comment]
@garygong Would appreciate it if you could give this a look when you have a chance.

**Comment by garygong** [comment]
The Combobox [Loading more options blueprint](https://www.lightningdesignsystem.com/components/combobox/#Loading-more-options) has a spinner without a container (`<div class="slds-spinner_container">`), but the `Spinner` component currently only supports rendering with a container.  I think you'd need an additional prop like `hasContainer` in order to implement the Combobox blueprint.  Do you want to include that in this PR too?

File: components/spinner/__docs__/storybook-stories.jsx [hunk 1]
```diff
@@ -18,6 +18,16 @@ const inverseContainer = (getStory) => (
 	<div style={inverseContainerStyle}>{getStory()}</div>
 );
 
+const inlineContainerStyle = {
```
Comment (garygong) [inline components/spinner/__docs__/storybook-stories.jsx:21]: The [Inlined blueprint](https://www.lightningdesignsystem.com/components/spinners/#Inlined) uses a container with: `<div class="slds-align_absolute-center" style="height:4rem">`.  I think we should stick with the same example here.

I think that might be related why the spinner in Storybook looks weirdly off screen for me:

image

After removing the `<div class="slds-spinner_container">` element and switching the container to use `<div class="slds-align_absolute-center" style="height:4rem">`, it looks a little better:

image

**Comment by jimmydief** [comment]
> The Combobox [Loading more options blueprint](https://www.lightningdesignsystem.com/components/combobox/#Loading-more-options) has a spinner without a container (`<div class="slds-spinner_container">`), but the `Spinner` component currently only supports rendering with a container. I think you'd need an additional prop like `hasContainer` in order to implement the Combobox blueprint. Do you want to include that in this PR too?

Thanks @garygong, happy to add it. I'd defer to you or @interactivellama on whether it makes sense to add that as a separate prop or to omit the container automatically when `isInline` is true.

**Comment by garygong** [comment]
Let's do a `hasContainer` prop.  Thanks!

File: components/combobox/private/menu.jsx [hunk 2]
```diff
@@ -508,16 +509,12 @@ const Menu = (props) => {
...
-							role="status"
-							className="slds-spinner slds-spinner_x-small slds-spinner_inline"
-						>
-							<span className="slds-assistive-text">
-								{props.assistiveText.loadingMenuItems}
-							</span>
-							<div className="slds-spinner__dot-a" />
-							<div className="slds-spinner__dot-b" />
-						</div>
+						<Spinner
```
Comment (jimmydief) [inline components/combobox/private/menu.jsx:512]: Can now use the inline spinner variant without container in the combobox itself.

**Comment by garygong** [comment]
LGTM!

**Comment by jimmydief** [comment]
Thanks @garygong! Any chance you can publish a release?

**Comment by garygong** [comment]
Yep, will do a new release this Friday.

**Pull request merged**